packages/sourcecred/src/core/ledger/policies/immediate.js

```diff
@@ -48,17 +50,43 @@ export function immediateReceipts(
     );
   }
 
-  const totalIntervals = identities[0].cred.length;
-  const shortTermCredPerIdentity = identities.map(({cred}) =>
-    cred
-      .slice(
-        cred.length - Math.min(policy.numIntervalsLookback, totalIntervals)
-      )
-      .reduce((sum, cred) => sum + cred, 0)
+  const intervalsBeforeEffective = credGrainView
+    .intervals()
+    .filter((interval) => interval.endTimeMs <= effectiveTimestamp);
+
+  const numIntervalsLookback = ((
```

Comment:
in your last PR, I offered a ternary operator solution that'd be nicer, more readable. Would you use that instead?